src/hex_http.erl

```diff
@@ -0,0 +1,27 @@
+-module(hex_http).
+-export([get/3]).
+
+-type adapter() :: module().
+-type status() :: non_neg_integer().
+-type headers() :: map().
+
+-callback get(URI :: string(), headers()) ->
+    {ok, status(), headers(), binary()} |
+    {error, term()}.
+-callback user_agent_string() -> string().
+
+-spec get(adapter(), string(), headers()) -> {ok, {status(), headers(), binary()}} | {error, term()}.
+get(Adapter, URI, Headers) ->
+    Headers2 = put_new("user-agent", user_agent(Adapter), Headers),
+    Adapter:get(URI, Headers2).
+
+user_agent(Adapter) ->
+    OtpVersion = erlang:system_info(otp_release),
+    UserAgentString = Adapter:user_agent_string(),
+    "hex_erl/0.1.0 " ++ UserAgentString ++ " (OTP/" ++ OtpVersion ++ ")".
```

Comment:
Can we add the client name and version here also?

Reply:
good idea 👍 

I'm not sure what's the best API to pass that information. Currently the API is: `hex_http:get(Adapter, URI, Headers)` and there's a `Adapter:user_agent_string/0` callback. Instead of callback we could pass data:

```erlang
Client = #{adapter => hex_http_httpc, user_agent_string => "(hex/0.18.0) (httpc) (Elixir/1.6.2)"},
hex_http:get(Client, URI, Headers)
```

which produces final UA:

```
hex_erl/0.1.0 (hex/0.18.0) (httpc) (Elixir/1.6.2) (OTP/20.1.2)
```

wdyt?

Reply:
Passing a user agent string sounds good.